src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -65,6 +68,47 @@ public class SocketTransportOptions
         /// </remarks>
         public bool UnsafePreferInlineScheduling { get; set; }
 
+        /// <summary>
+        /// A function used to create a new <see cref="Socket"/> to listen with. If
+        /// not set, <see cref="CreateDefaultListenSocket" /> is called instead.
+        /// </summary>
+        public Func<EndPoint, Socket>? CreateListenSocket { get; set; }
```

Comment:
Can we preinitialize this with CreateDefaultListenSocket and make it non-nullable?